### PR TITLE
*: document client package vendoring guide

### PR DESCRIPTION
**Pull request by gyuho**
Related https://github.com/coreos/etcd/issues/4904.

### CODE REVIEW

File: client/README.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 etcd/client is the Go client library for etcd.
 
+etcd uses `vendor` directory to manage external dependencies. If `client` is imported
+outside of etcd, the project needs its own vendoring to prevent package conflicts. For more
+detail, please read [Go 1.5 vendor](https://golang.org/s/go15vendor).
```
Comment (xiang90) [inline client/README.md]: also add a link to https://golang.org/cmd/go/#hdr-Vendor_Directories?

File: client/README.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 etcd/client is the Go client library for etcd.
 
+etcd uses `vendor` directory to manage external dependencies. If `client` is imported
+outside of etcd, the project needs its own vendoring to prevent package conflicts. For more
```
Comment (xiang90) [inline client/README.md]: Simply copy client to `vendor` dir in your application or use tools like goDep to manage your dependency.

**Comment by gyuho** [comment]
@xiang90 PTAL. Thanks!

**Comment by xiang90** [comment]
LGTM. /cc @heyitsanthony can you give this a quick look too?

File: client/README.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 etcd/client is the Go client library for etcd.
 
+etcd uses `vendor` directory to manage external dependencies. If `client` is imported
```
Comment (heyitsanthony) [inline client/README.md]: uses go's `vendor` directory

File: client/README.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
...
+outside of etcd, simply copy `client` to `vendor` directory or use tools like godep to
```
Comment (heyitsanthony) [inline client/README.md]: to the `vendor` directory

File: clientv3/README.md [hunk 5]
```diff
@@ -40,6 +40,11 @@ if err != nil {
 // use the response
 ```
 
+etcd uses `vendor` directory to manage external dependencies. If `clientv3` is imported
```
Comment (heyitsanthony) [inline clientv3/README.md]: uses go's vendor directory

File: clientv3/README.md [hunk 6]
```diff
@@ -40,6 +40,11 @@ if err != nil {
...
+outside of etcd, simply copy `clientv3` to `vendor` directory or use tools like godep to
```
Comment (heyitsanthony) [inline clientv3/README.md]: to the `vendor` directory

**Comment by heyitsanthony** [comment]
lgtm after fixups

**Pull request merged**